cmake/detect-intrinsics.cmake

```diff
@@ -1,15 +1,25 @@
 # detect-intrinsics.cmake -- Detect compiler intrinsics support
 # Licensed under the Zlib license, see LICENSE.md for details
 
-macro(check_acle_intrinsics)
-    if(NOT NATIVEFLAG)
-        set(ACLEFLAG "-march=armv8-a+crc")
+macro(check_acle_compiler_flag)
+    if(NOT NATIVEFLAG AND NOT HAVE_ACLE_INTRIN)
+        set(ACLEFLAG "-march=armv8-a+crc" CACHE INTERNAL "Compiler option to enable ACLE support")
     endif()
     # Check whether compiler supports ACLE flag
     set(CMAKE_REQUIRED_FLAGS "${ACLEFLAG}")
     check_c_source_compiles(
         "int main() { return 0; }"
         HAVE_ACLE_INTRIN FAIL_REGEX "not supported")
+    if(NOT NATIVEFLAG AND NOT HAVE_ACLE_INTRIN)
+        set(ACLEFLAG "-march=armv8-a+crc+simd" CACHE INTERNAL "Compiler option to enable ACLE support" FORCE)
```

Comment:
Is this so `ACLEFLAG` can be overwritten in the command line or for something else?

Reply:
It's to allow cmake reconfigure itself if `CMakeLists.txt` gets updated between builds.